### ISSUE TITLE
Does yolo differentiate between classes based on shape or color?

### ISSUE THREAD

**Issue by nidhimittalhada** [thtrieu/darkflow#783]
Hi all,

**This is a fundamental question i am struggling to find answer of everywhere.**

I was asked to solve a computer vision problem of detecting a type of carton box in the image. 
if its a white box or a brown box.. when boxes have same shape. I had two classes whitebox, brownbox.

I tried yolov1 with tiny version on this problem. I trained my model on 1500 approx images for each class. My model's loss reduced to 0.04 when i stopped training.

Now, inference on test image is very bad. 
It mostly detecting wrong class. To white box it mostly says a brownbox. 
Though bounding box predicted coordinates looks to be correct, but class prediction on test images is too bad.

I am really confused when training loss went upto 0.04 how can model predict wrong class almost on every image which i test on.

This is my config settings
[detection]
classes=2
coords=4
rescore=1
side=7
num=2
softmax=0
sqrt=1
jitter=.2
object_scale=1
noobject_scale=.5
class_scale=1
coord_scale=5

I am thinking class_scale is 1 should i increase it to train so that class prediction comes out to be correct ?

OR IS there fundamental mistake in my concept ? can yolo understand difference between two objects of same shape but different color?

Any kind of help in this regard, any kind of help from your experience will be reallllly very much appreciated ..

**Comment by AbhishekSingl** [comment]
Hi Nidhi,
It's been 2 years since you post above question. Can you please answer to this query in case you've figured it out? I'm struggling with the same question - whether YOLO differentiate between classes based on shape or color?

**Comment by Kiwi97** [comment]
https://github.com/pjreddie/darknet/issues/1413#issuecomment-459711379  answer
